inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/KnowledgeBaseServiceImpl.java

```diff
@@ -1184,4 +1197,57 @@ public File readKbFileFromClassPathResource(String aLocation) throws IOException
         return kbFile;
     }
 
+
+    private IndexWriter getIndexWriter(KnowledgeBase aKb) throws IOException
+    {
+        Analyzer analyzer = new StandardAnalyzer();
+        Directory directory = FSDirectory
+            .open(new File(luceneIndexDir, aKb.getRepositoryId()).toPath());
+        return new IndexWriter(directory, new IndexWriterConfig(analyzer));
+    }
+
+    @Override
+    public void indexLocalKb(KnowledgeBase aKb)
+    {
+        try {
+            IndexWriter indexWriter = getIndexWriter(aKb);
+
+            try (RepositoryConnection conn = getConnection(aKb)) {
+                RepositoryResult<Statement> stmts = RdfUtils
+                    .getStatementsSparql(conn, null, aKb.getLabelIri(), null, Integer.MAX_VALUE,
+                        false, null);
+                while (stmts.hasNext()) {
+                    Statement stmt = stmts.next();
+                    String id = stmt.getSubject().stringValue();
+                    String label = stmt.getObject().stringValue();
+                    indexCreatedConcept(id, label, indexWriter);
+                }
+                conn.commit();
+            }
+        }
+        catch (IOException e) {
+            log.error("Could not index local KB.", e);
+        }
+    }
+
+    private void indexCreatedConcept(String aId, String aLabel, IndexWriter aIndexWriter)
```

Comment:
Did you debug if `LuceneSailConnection.commit()` properly updates the index with the stored triples?

Reply:
Where is this function used?